iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/pipe/task/PipeTaskCoordinator.java

```diff
@@ -102,23 +87,36 @@ public TSStatus dropPipe(String pipeName) {
   }
 
   public TShowPipeResp showPipes(TShowPipeReq req) {
-    return ((PipeTableResp)
-            configManager.getConsensusManager().read(new ShowPipePlanV2()).getDataset())
-        .filter(req.whereClause, req.pipeName)
-        .convertToTShowPipeResp();
+    // Should take read lock here to avoid writes between read() and convertTo...()
+    pipeTaskInfo.acquireReadLock();
+    final TShowPipeResp resp =
+        ((PipeTableResp)
+                configManager.getConsensusManager().read(new ShowPipePlanV2()).getDataset())
+            .filter(req.whereClause, req.pipeName)
+            .convertToTShowPipeResp();
+    pipeTaskInfo.releaseReadLock();
+    return resp;
```

Comment:
```suggestion
    // Should take read lock here to avoid writes between read() and convertTo...()
    pipeTaskInfo.acquireReadLock();
    try {
      return ((PipeTableResp)
                configManager.getConsensusManager().read(new ShowPipePlanV2()).getDataset())
              .filter(req.whereClause, req.pipeName)
              .convertToTShowPipeResp();
    } finally {  
      pipeTaskInfo.releaseReadLock();
    }
```